frontend/src/components/elements/ArrowVegaLiteChart/ArrowVegaLiteChart.tsx

```diff
@@ -167,7 +167,8 @@ export class ArrowVegaLiteChart extends PureComponent<PropsWithHeight, State> {
       prevSpec !== spec ||
       prevTheme !== theme ||
       prevProps.width !== this.props.width ||
-      prevProps.height !== this.props.height
+      prevProps.height !== this.props.height ||
+      prevProps.element.vegaLiteTheme !== this.props.element.vegaLiteTheme
```

Comment:
This line is the actual fix or? the renaming itself is more optional?

Reply:
Yes this is the line that is the actual fix. The renaming itself is more optional but I just thought it was confusing to have `prevTheme !== theme` and then we have `prevProps.element.theme !== this.props.element.theme` so I changed it

Reply:
Ahh, I see. Yes, the naming of the app theme and chart theme can be a bit confusing. Using  `vegaLiteTheme` is ok, but it's also not super optimal since this is the only parameter diverging from the actual proto parameter namings. But I don't really have any better solution to offer.

I think the actual best solution would be to transform this into a functional component and use conditional effects for this. This is a bit more work, but maybe might be worth it to take on soon. Especially since vega lite is a quite an important component with all the built-in charts.